app/controllers/api/categories_controller.rb

```diff
@@ -8,13 +8,13 @@ def show
     end
 
     def edit_resource(type, id, data = {})
-      raise BaseController::Forbidden if Category.find(id).read_only?
+      raise Forbidden if Category.find(id).read_only?
```

Comment:
`Category.where(:id => id).pluck(:read_only?)`

Only different is that the find will blown up if the record was not found, but I don't think you should bring back the whole category if you only need the one flag.